rpcs3/Emu/RSX/Overlays/overlay_edit_text.cpp

```diff
@@ -28,10 +28,69 @@ namespace rsx
 				break;
 			}
 			case up:
+			{
+				if (!caret_position)
+				{
+					// Special case - caret is at the very beginning
+					break;
+				}				
+				const u64 current_line_start = text.rfind('\n', caret_position - 1) + 1;
+				if (current_line_start == std::string::npos)
+				{
+					// This is the first line, so we can't move upwards
+					break;
+				}				
+				const u64 caret_pos_in_line = caret_position - current_line_start;
+				const u64 prev_line_end    = current_line_start - 1;
+				const u64 prev_line_start  = text.rfind('\n', prev_line_end - 1); // - 1 is ignore '\n' and begin search before it
+				const u64 prev_line_length = prev_line_end - prev_line_start;
+				// TODO : Save caret position to some kind of buffer, so after switching back and forward, caret would be on initial position
+				if (prev_line_length < caret_pos_in_line)
+				{
+					// Previous line is shorter than current caret position, so caret moves to the very end of the previous line
+					caret_position = prev_line_end;
+				}
+				else
+				{
+					// Previous line is longer than current, so caret should offset from prev_line_beginning the exact same distance as in current line
+					caret_position = prev_line_start + caret_pos_in_line;
+				}
+				refresh();
+				break;
+			}
 			case down:
-				// TODO
+			{
+				if (caret_position == text.length())
```

Comment:
Text.length or text.length -1 ?
Idk. Maybe needs investigating

Reply:
if I check it with == it actually means, that caret is after last existing char = very end.

Reply:
-1 would mean "place caret before last char"

Reply:
Yes. I can see that now